File: packages/mlforge_sdk/mlforge_sdk/projects.py
from __future__ import annotations

from typing import List, Optional

from pydantic import BaseModel, ConfigDict

from .http import HttpClient


class Project(BaseModel):
    model_config = ConfigDict(extra="ignore")
    id: str
    name: str
    path: str
    created_at: str
    last_opened: str
    status: str = "idle"


class ProjectsClient:
    def __init__(self, http: HttpClient):
        self._http = http

    def list(self, *, limit: int = 200, offset: int = 0) -> List[Project]:
        data = self._http.get("/projects", params={"limit": limit, "offset": offset})
        return [Project(**p) for p in data]

    def upsert(self, project: Project) -> Project:
        data = self._http.post("/projects", json_body=project.model_dump())
        return Project(**data)

    def open(self, project_id: str) -> None:
        self._http.post(f"/projects/{project_id}/open")

    def delete(self, project_id: str) -> None:
        self._http.delete(f"/projects/{project_id}")

File: packages/mlforge_js_sdk/src/projects.ts
import type { HttpClient } from "./http.js";

export interface Project {
  id: string;
  name: string;
  path: string;
  created_at: string;
  last_opened: string;
  status?: string;
}

export interface ProjectCreateOptions {
  name: string;
}

export class ProjectsClient {
  constructor(private http: HttpClient) {}

  async list(opts?: { limit?: number; offset?: number }): Promise<Project[]> {
    return this.http.get<Project[]>("/projects", {
      limit: opts?.limit ?? 200,
      offset: opts?.offset ?? 0,
    });
  }

  async create(opts: ProjectCreateOptions): Promise<Project> {
    return this.http.post<Project>("/projects", opts);
  }

  async get(projectId: string): Promise<Project> {
    return this.http.get<Project>(`/projects/${projectId}`);
  }

  async open(projectId: string): Promise<void> {
    return this.http.post<void>(`/projects/${projectId}/open`);
  }

  async delete(projectId: string): Promise<void> {
    return this.http.delete<void>(`/projects/${projectId}`);
  }
}
